File: baekjoon-online-judge/normal/01034.py
from collections import defaultdict
from sys import stdin

input = stdin.readline

if __name__ == '__main__':
    # Input
    N, M = map(int, input().split())
    board = [input().strip() for _ in range(N)]
    K = int(input())

    # Solve
    temp = defaultdict(int)
    for info in board:
        temp[info] += 1

    res = 0
    for key, value in temp.items():
        num = key.count('0')
        if K >= num and (K - num) % 2 == 0:
            res = max(res, value)

    # Output
    print(res)

File: baekjoon-online-judge/normal/01034.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		// Input
		String[] in = br.readLine().split(" ");
		final int N =  Integer.parseInt(in[0]);
		final int M =  Integer.parseInt(in[1]);

		char[][] board = new char[N][M];
		String[] boardStr = new String[N];
		for (int i = 0; i < N; i++) {
			boardStr[i] = br.readLine();
			board[i] = boardStr[i].toCharArray();
		}

		final int K = Integer.parseInt(br.readLine());

		// Solve
		int result = 0;
		for (int row1 = 0; row1 < N; ++row1) {
			int zeroCnt = 0;
			for (int j = 0; j < M; ++j) {
				if (board[row1][j] == '0') {
					++zeroCnt;
				}
			}

			int max = 0;
			if (zeroCnt <= K && zeroCnt % 2 == K % 2) {
				for (int row2 = 0; row2 < N; ++row2) {
					if (boardStr[row1].equals(boardStr[row2])) {
						++max;
					}
				}
			}
			result = Math.max(result, max);
		}

		// Output
		System.out.println(result);
	}
}